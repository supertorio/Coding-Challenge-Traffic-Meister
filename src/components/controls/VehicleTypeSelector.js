import React from 'react';
import {connect} from "react-redux";

import {getCurrentVehicleType, getAvailableVehicleTypes} from "../../reducers/index-reducer";
import {changeVehicleTypeSelection} from "../../actions/meister-form-actions";
import Dropdown from "./Dropdown";

const VehicleTypeSelector = (props) => {
    return (
        <div className="Selector VehicleTypeSelector">
            <Dropdown label="Vehicle Type"
                      value={props.selectedType}
                      options={props.vehicles}
                      onChange={props.changeVehicleTypeSelection} />
        </div>
    );
};

const mapStateToProps = state => {
    return {
        vehicles: getAvailableVehicleTypes(state),
        selectedType: getCurrentVehicleType(state),
    };
};

const mapDispatchToProps = {
    changeVehicleTypeSelection
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(VehicleTypeSelector);
