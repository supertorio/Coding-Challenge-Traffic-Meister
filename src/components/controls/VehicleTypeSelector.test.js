import React from 'react';
import {VehicleTypeSelector} from './VehicleTypeSelector';
import {List} from 'immutable';

describe('<VehicleTypeSelector />', () => {

    let types, selectedType;

    beforeEach(()=> {
        types = new List(['car', 'train', 'airplane']);
        selectedType = types.get[0];
    });

    it('should match the snapshot', () => {
        const wrapper = shallow(
            <VehicleTypeSelector vehicles={types} selectedType={selectedType} changeVehicleTypeSelection={() => true} />,
        );
        expect(wrapper).toMatchSnapshot();
    });

});
