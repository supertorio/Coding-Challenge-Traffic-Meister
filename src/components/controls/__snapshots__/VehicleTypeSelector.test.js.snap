// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VehicleTypeSelector /> should match the snapshot 1`] = `ShallowWrapper {}`;
